src/common/memory_manage.h

```diff
@@ -1,96 +1,132 @@
     }
 
     // Allocate the table and fill it with random floating point values between
     // -1.0 and 1.0
-    void rand_alloc_float(size_t dim1, size_t dim2, size_t aligned_bytes)
+    void rand_alloc_float(
+        size_t dim1, size_t dim2, Agora_memory::Alignment_t alignment)
     {
         std::default_random_engine generator;
         std::uniform_real_distribution<float> distribution(-1.0, 1.0);
 
-        malloc(dim1, dim2, aligned_bytes);
-        auto* base = reinterpret_cast<float*>(data);
-        for (size_t i = 0; i < dim1 * dim2; i++) {
+        assert(sizeof(T) >= sizeof(float));
+        this->malloc(dim1, dim2, alignment);
+        auto* base = reinterpret_cast<float*>(this->data_);
+        for (size_t i = 0; i < (dim1 * dim2); i++) {
             base[i] = distribution(generator);
         }
     }
 
     // Allocate the table and fill it with random complex floating point values
     // between -1.0 and 1.0
-    void rand_alloc_cx_float(size_t dim1, size_t dim2, size_t aligned_bytes)
+    void rand_alloc_cx_float(
+        size_t dim1, size_t dim2, Agora_memory::Alignment_t alignment)
     {
         std::default_random_engine generator;
         std::uniform_real_distribution<float> distribution(-1.0, 1.0);
 
-        malloc(dim1, dim2, aligned_bytes);
-        auto* base = reinterpret_cast<float*>(data);
-        for (size_t i = 0; i < dim1 * dim2 * 2; i++) {
-            base[i] = distribution(generator);
+        assert(sizeof(T) >= sizeof(std::complex<float>));
+        this->malloc(dim1, dim2, alignment);
+        auto* base = reinterpret_cast<std::complex<float>*>(this->data_);
+        for (size_t i = 0; i < (dim1 * dim2); i++) {
+            base[i] = { distribution(generator), distribution(generator) };
         }
     }
 
-    bool is_allocated() { return dimension > 0; }
+    bool is_allocated(void) { return (this->data_ != nullptr); }
 
     void free(void)
     {
-        std::free(data);
-        dimension = 0;
-        data = NULL;
+        if (this->data_ != nullptr) {
+            std::free(this->data_);
+        }
+        this->dim2_ = 0;
+        this->dim1_ = 0;
+        this->data_ = nullptr;
     }
 
-    T* operator[](int dim1) { return (T*)((char*)data + dim1 * dimension); }
+    T* operator[](size_t dim1)
+    {
+        assert(this->dim1_ > dim1);
+        return (this->data_ + (dim1 * this->dim2_));
+    }
 };
 
 template <typename T, typename U>
-static void alloc_buffer_1d(T** buffer, U dim, int aligned_bytes, int init_zero)
+static void alloc_buffer_1d(
+    T** buffer, U dim, Agora_memory::Alignment_t alignment, int init_zero)
 {
-    aligned_bytes = (aligned_bytes) / 32 * 32;
-    *buffer = (T*)aligned_alloc(aligned_bytes, dim * sizeof(T));
-    if (init_zero)
-        memset(*buffer, 0, dim * sizeof(T));
+    size_t size = dim * sizeof(T);
+    *buffer
+        = static_cast<T*>(Agora_memory::padded_aligned_alloc(alignment, size));
```

Comment:
`static_cast` and `reinterpret_cast` are both used in memory allocation across multiple files. Shall we pick one and make it consistent?

Reply:
Yes,  from my brief internet search I think static_cast would be most correct but I am not 100% sure.    Let me know and I'll update all of them consistently. 

https://embeddedartistry.com/blog/2017/03/15/c-casting-or-oh-no-they-broke-malloc/

Reply:
Thanks for sharing the articles. I think `static_cast` should be used for all cases of memory allocation with a return type of `void *`. For other casts, we can keep them as is.